csrc/scheduler/tools/maxinfo_propagator.cpp

```diff
@@ -470,7 +471,12 @@ bool SetSelector::allowP2C(TensorView* from, TensorView* to) {
 }
 
 bool SetSelector::allowSibling(TensorView* from, TensorView* to) {
-  return true;
+  // Allow propagation between siblings if they are uniform to avoid errors in
```

Comment:
We should probably do this change in `MaxInfoSpanningTree::compute_spanning_tree` itself. `Selector` is optional, and if not specified, it allows to propagate to siblings by default.